test-basic/basePath-test.js

```diff
@@ -0,0 +1,92 @@
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const testconfig = require('../etc/test-config.js');
+const marklogic = require('../');
+let assert = require('assert');
+let writeObject = {
+    uri: '/write/string1.json',
+    contentType: 'application/json',
+    content: '{"key1":"value 1"}'
+};
+
+describe('basePath tests', function() {
+
+    it('basePath without slash(s)', function(done){
+        testconfig.restWriterConnectionWithBasePath.basePath = 'invalid';
+        const dbWriter = marklogic.createDatabaseClient(testconfig.restWriterConnectionWithBasePath);
+        dbWriter.documents.write(writeObject)
+            .result(function(response){})
+            .catch(err=>
+            {
+                assert(err.toString().includes('basePath invalid/v1/documents'));
+                done();
+            });
+    });
+
+    it('basePath with slash', function(done){
+        testconfig.restWriterConnectionWithBasePath.basePath = '/invalid';
+        const dbWriter = marklogic.createDatabaseClient(testconfig.restWriterConnectionWithBasePath);
+        dbWriter.documents.write(writeObject)
+            .result(function(response){})
+            .catch(err=>
+            {
+                assert(err.toString().includes('basePath invalid/v1/documents'));
+                done();
+            });
+    });
+
+    it('basePath with trailing slash', function(done){
+        testconfig.restWriterConnectionWithBasePath.basePath = 'invalid/';
+        const dbWriter = marklogic.createDatabaseClient(testconfig.restWriterConnectionWithBasePath);
+        dbWriter.documents.write(writeObject)
+            .result(function(response){})
+            .catch(err=>
+            {
+                assert(err.toString().includes('basePath invalid/v1/documents'));
+                done();
+            });
+    });
+
+    it('basePath with starting and trailing slashes', function(done){
+        testconfig.restWriterConnectionWithBasePath.basePath = '/invalid/';
+        const dbWriter = marklogic.createDatabaseClient(testconfig.restWriterConnectionWithBasePath);
+        dbWriter.documents.write(writeObject)
+            .result(function(response){})
+            .catch(err=>
+            {
+                assert(err.toString().includes('basePath invalid/v1/documents'));
+                done();
+            });
+    });
+
+    it('basePath with multiple starting and trailing slashes', function(done){
+        testconfig.restWriterConnectionWithBasePath.basePath = '//invalid//';
+        const dbWriter = marklogic.createDatabaseClient(testconfig.restWriterConnectionWithBasePath);
```

Comment:
This test currently fails because we remove only one slash before and after the word "invalid" and hence the basePath it tries to connect is - "/invalid//v1/documents"

Reply:
Updated this test as per the discussion in this morning's meeting. Thanks.!